soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -12703,16 +12536,20 @@ void func_8084DFAC(GlobalContext* globalCtx, Player* this) {
 }
 
 s32 func_8084DFF4(GlobalContext* globalCtx, Player* this) {
-    GetItemEntry* giEntry;
+    GetItemEntry giEntry;
     s32 temp1;
     s32 temp2;
 
-    if (this->getItemId == GI_NONE) {
+    if (this->getItemId == GI_NONE && this->getItemEntry.objectId == OBJECT_INVALID) {
         return 1;
     }
 
     if (this->unk_84F == 0) {
-        giEntry = &sGetItemTable[this->getItemId - 1];
+        if (this->getItemEntry.objectId == OBJECT_INVALID) {
+            giEntry = ItemTable_Retrieve(this->getItemId - 1);
```

Comment:
noting another `-1` in case that's a thing that we end up wanting to change

Reply:
Addressed by 69086f58c2b6bb00a273eefed4744ad456bba6ba